app/controllers/teachers/registrations_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class Teachers::RegistrationsController < Devise::RegistrationsController
+  # GET /resource/sign_up
+  # def new
+  #   super
+  # end
+
+  # POST /resource
+  def create
```

Comment:
This now looks like perfectly normal devise registrations controller. Can you please try removing it, removing the special route for it and seeing if devise itself is good enough to do the magic for you? :) Might turn out you don't even need this controller at all :)

Reply:
Removed controller and custom rout and it still works. How it is even possible?

Reply:
About coverage/* - did You mean to remove them altogether from .gitignore?? I am a bit perplexed with this one. Sorry :(

Reply:
> Removed controller and custom rout and it still works. How it is even possible?

Devise does pretty good job with the defaults :) If there's no need to override them - don't :)